src/hooks/useFetchPriceHistories/useFetchPriceHistories.ts

```diff
@@ -13,13 +14,20 @@ type UseFetchPriceHistories = (args: UseFetchPriceHistoriesArgs) => void
 
 export const useFetchPriceHistories: UseFetchPriceHistories = ({ assetIds, timeframe }) => {
   const dispatch = useAppDispatch()
+  const symbol = useAppSelector(selectSelectedCurrency)
+
+  const { findPriceHistoryByAssetId, findPriceHistoryByFiatSymbol } = marketApi.endpoints
   useEffect(
     () =>
       assetIds.forEach(assetId =>
-        dispatch(marketApi.endpoints.findPriceHistoryByAssetId.initiate({ assetId, timeframe })),
+        dispatch(findPriceHistoryByAssetId.initiate({ assetId, timeframe })),
       ),
     // assetIds ref changes, prevent infinite render
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [JSON.stringify(assetIds), dispatch, timeframe],
+    [assetIds, dispatch, timeframe],
+  )
+  useEffect(
+    () => void dispatch(findPriceHistoryByFiatSymbol.initiate({ symbol, timeframe })),
+    [dispatch, findPriceHistoryByFiatSymbol, symbol, timeframe],
```

Comment:
Would prefer seeing curly braces rather than an obscure casting syntax (also you shouldn't return for a reason, the return value of `useEffect` is actually used and checked for: if it's a function, it will be used as a cleanup function which is why you were seeing type errors)

```suggestion
useEffect(() => {
    dispatch(findPriceHistoryByFiatSymbol.initiate({ symbol, timeframe }))
  }, [dispatch, findPriceHistoryByFiatSymbol, symbol, timeframe]
```